cadence/contracts/FlowYieldVaultsScheduler.cdc

```diff
@@ -177,50 +180,75 @@ access(all) contract FlowYieldVaultsScheduler {
-                                    "scanForStuck": scanForStuck
-                                }
-
-                                let selfTxn <- FlowTransactionScheduler.schedule(
-                                    handlerCap: supervisorCap!,
-                                    data: nextData,
-                                    timestamp: nextTimestamp,
-                                    priority: priority,
-                                    executionEffort: executionEffort,
-                                    fees: <-fees
-                                )
-
-                                emit SupervisorRescheduled(
-                                    scheduledTransactionID: selfTxn.id,
-                                    timestamp: nextTimestamp
-                                )
-
-                                destroy selfTxn
-                            }
-                        }
+                self.scheduleNextRecurringExecution(
+                    recurringInterval: interval,
+                    priority: priority,
+                    priorityRaw: priorityRaw,
+                    executionEffort: executionEffort,
+                    scanForStuck: scanForStuck
+                )
+            }
+        }
+
+        /// Self-reschedules the Supervisor for perpetual operation.
+        ///
+        /// This function handles the scheduling of the next Supervisor execution,
+        /// including fee estimation, withdrawal, and transaction scheduling.
+        ///
+        /// @param recurringInterval: The interval in seconds until the next execution
+        /// @param priority: The priority level for the scheduled transaction
+        /// @param priorityRaw: The raw priority value (UInt8) for data serialization
+        /// @param executionEffort: The execution effort estimate for the transaction
+        /// @param scanForStuck: Whether to scan for stuck yield vaults in the next execution
+        access(Schedule) fun scheduleNextRecurringExecution(
+            recurringInterval: UFix64,
+            priority: FlowTransactionScheduler.Priority,
+            priorityRaw: UInt8,
+            executionEffort: UInt64,
+            scanForStuck: Bool
+        ) {
+            let nextTimestamp = getCurrentBlock().timestamp + recurringInterval
+            let supervisorCap = FlowYieldVaultsSchedulerRegistry.getSupervisorCap()
+
+            if supervisorCap == nil || !supervisorCap!.check() {
+                return
+            }
+
+            let est = FlowYieldVaultsScheduler.estimateSchedulingCost(
+                timestamp: nextTimestamp,
+                priority: priority,
+                executionEffort: executionEffort
+            )
+            let baseFee = est.flowFee ?? FlowYieldVaultsScheduler.MIN_FEE_FALLBACK
+            let required = baseFee * FlowYieldVaultsScheduler.FEE_MARGIN_MULTIPLIER
+
+            if let vaultRef = self.feesCap.borrow() {
```

Comment:
An alternative to no-op could be to emit a failure case event for more robust offchain monitoring. This would be an upgradeable change so a non-blocker.